### ISSUE TITLE
`redefined-loop-name` warns on tuple unpacking swap

### ISSUE THREAD

**Issue by injust** [astral-sh/ruff#22218]
### Summary

Using tuple unpacking to swap variables is common, but PLW2901 warns because it's technically overwriting the for-loop variables:

https://play.ruff.rs/1be1a287-f128-4840-816a-21818db565bb

```python
from itertools import pairwise

def test(nums: list[int]) -> None:
    for a, b in pairwise(nums):
        if b > a:
            a, b = b, a
        print(a, b)
```

Perhaps the rule should be suppressed when variables are being swapped, as long as they were both defined in the same loop.

**Comment by ntBre** [comment]
Thanks for the report! I'm not sure this is worth a general exception to the rule, but I'm open to other opinions.

cc @amyreese

**Comment by amyreese** [comment]
tbh, I can't remember ever seeing loop variables swapped like that, since they'll just be reset on the next iteration. I'd have to see meaningful usage in the wild to think the exception is worth it.

530 existing suppressions of this rule in the wild that do multiple assignment, including false positive hits because regex: https://github.com/search?q=%2F.%2B%2C.%2B%3D.*PLW2901%2F&type=code
